### ISSUE TITLE
Missing config default_value in config_fn on composite_solid

### ISSUE THREAD

**Issue by kinghuang** [dagster-io/dagster#1608]
I have a composite solid with an optional config named `prefix`, which is used in config_fn as the value for an inner solid's config. Since the config is optional, I expect the default value to be used if I don't specify a value in an environment. But, I get a KeyError on `prefix` instead (i.e., the config dict passed in config_fn doesn't have the `prefix` key at all).

Here's the composite solid and environment.

```python
@composite_solid(
    config_fn=lambda _, cfg: {'prefix_value': {'config': {'prefix': cfg['prefix']}}},
    config={'prefix': Field(str, is_optional=True, default_value='_id_')}
)
def prefix_id(id):
    return prefix_value(id)
```

```yaml
solids:
  prefix_id:
    config: {}
    inputs:
      id:
        value: "12345"
```

The empty config is due to #1607.

Here is the exception.

```
Exception occurred during execution of user config mapping function <lambda> defined by solid prefix_id from definition prefix_id at path root:solids:prefix_id:
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/dagster/core/types/evaluator/evaluation.py", line 252, in _evaluate_composite_solid_config
    frozendict(context.config_value.get('config')),
  File "/project/spendanalytics/nlp/normalize.py", line 17, in <lambda>
    config_fn=lambda _, cfg: {'prefix_value': {'config': {'prefix': cfg['prefix']}}},
KeyError: 'prefix'
```

Specifying a value for the `prefix` config works, but rather defeats the default value. I expect the config passed to `config_fn` to have a `prefix` key with the value `_id_` from the composite's declared config.

```yaml
solids:
  prefix_id:
    config:
      prefix: _foo_
    inputs:
      id:
        value: "12345"

```

**Comment by natekupp** [comment]
Hey King Chung, thanks for reporting! This is indeed a real issue with the way we are currently applying default values from `Field` definitions. 

Unfortunately I think it's going to take us a bit longer to fix, because the config traversal system needs a more holistic rework to support fixing defaults application. I added a note discussing the next steps on that front here: https://github.com/dagster-io/dagster/issues/692#issuecomment-509386880

In the interim, I saw @mgasner was discussing presets with you on Slack, let us know if that is a good fit for your problem. I'll also follow up directly on Slack.
